gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -899,20 +889,55 @@ def create_final_freq_ht(ht: hl.Table) -> hl.Table:
     """
     Create final freq Table with only desired annotations.
 
-    Drop the following annotations:
-        - 'age_high_ab_hist', all annotations from the split VDSs....
-
-    Rename the following annotations:
-        - 'ab_adjusted_freq' -> 'freq'
-        - decide if we want to store uncorrected? Probably,just rename it.
+    .. note::
 
-    Filter the following from the 'freq' field:
-        - gatk versions
+        `pre_adjustment_freq` is the frequency annotation before any high AB het
+        adjustment. This value will be the same as freq for sites that have an AF below
+        the AF adjustment threshold.
 
     :param ht: Hail Table containing all annotations.
-    :return: Hail Table with only desired annotations.
+    :return: Hail Table with final annotations.
     """
-    raise NotImplementedError("Creation of final freq Table is not implemented yet.")
+    logger.info("Dropping gatk_version from freq ht array annotations...")
+    freq_meta, array_exprs = filter_arrays_by_meta(
+        ht.freq_meta,
+        {
+            "freq": ht.ab_adjusted_freq,
+            "freq_meta_sample_count": ht.index_globals().freq_meta_sample_count,
+            "high_ab_hets_by_group": ht.high_ab_hets_by_group,
+            "pre_adjustment_freq": ht.freq,
+        },
+        items_to_filter=["gatk_version"],
+        keep=False,
+    )
+
+    ht = ht.select(
+        freq=array_exprs["freq"],
+        faf=ht.faf,
+        grpmax=ht.grpmax,
+        inbreeding_coeff=ht.InbreedingCoeff,
+        histograms=hl.struct(
+            **{
+                "qual_hists": ht.ab_adjusted_qual_hists,
+                "raw_qual_hists": ht.ab_adjusted_raw_qual_hists,
+                "age_hists": ht.ab_adjusted_age_hists,
+            },
+        ),
+        pre_adjustment_freq=array_exprs["pre_adjustment_freq"],
+        high_ab_hets_by_group=array_exprs["high_ab_hets_by_group"],
+    )
+
+    ht = ht.select_globals(
+        downsamplings=ht.index_globals().downsamplings,
+        freq_meta=freq_meta,
+        freq_index_dict=make_freq_index_dict_from_meta(freq_meta),
+        freq_meta_sample_count=array_exprs["freq_meta_sample_count"],
+        faf_meta=ht.index_globals().faf_meta,
+        faf_index_dict=ht.index_globals().faf_index_dict,
+        age_distribution=ht.index_globals().age_distribution,
+    )
```

Comment:
Slight preference for this, but don't really care:

```suggestion
    g = ht.index_globals()
    ht = ht.select_globals(
        downsamplings=g.downsamplings,
        freq_meta=freq_meta,
        freq_index_dict=make_freq_index_dict_from_meta(freq_meta),
        freq_meta_sample_count=array_exprs["freq_meta_sample_count"],
        faf_meta=g.faf_meta,
        faf_index_dict=g.faf_index_dict,
        age_distribution=g.age_distribution,
    )
```